test/conformance/kernel/urKernelGetInfo.cpp

```diff
@@ -3,70 +3,129 @@
 // See LICENSE.TXT
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
+#include "ur_api.h"
 #include <uur/fixtures.h>
 
-using urKernelGetInfoTest = uur::urKernelTestWithParam<ur_kernel_info_t>;
+using urKernelGetInfoTest = uur::urKernelTest;
+UUR_INSTANTIATE_KERNEL_TEST_SUITE_P(urKernelGetInfoTest);
 
-UUR_TEST_SUITE_P(
-    urKernelGetInfoTest,
-    ::testing::Values(UR_KERNEL_INFO_FUNCTION_NAME, UR_KERNEL_INFO_NUM_ARGS,
-                      UR_KERNEL_INFO_REFERENCE_COUNT, UR_KERNEL_INFO_CONTEXT,
-                      UR_KERNEL_INFO_PROGRAM, UR_KERNEL_INFO_ATTRIBUTES,
-                      UR_KERNEL_INFO_NUM_REGS),
-    uur::deviceTestWithParamPrinter<ur_kernel_info_t>);
+TEST_P(urKernelGetInfoTest, FunctionName) {
+    auto property_name = UR_KERNEL_INFO_FUNCTION_NAME;
+    size_t property_size = 0;
+
+    ASSERT_SUCCESS(
+        urKernelGetInfo(kernel, property_name, 0, nullptr, &property_size));
+    ASSERT_GT(property_size, 0U);
 
-using urKernelGetInfoSingleTest = uur::urKernelExecutionTest;
-UUR_INSTANTIATE_KERNEL_TEST_SUITE_P(urKernelGetInfoSingleTest);
+    std::vector<char> property_value(property_size);
+    ASSERT_SUCCESS(urKernelGetInfo(kernel, property_name, property_size,
+                                   property_value.data(), nullptr));
+}
 
-TEST_P(urKernelGetInfoTest, Success) {
-    auto property_name = getParam();
+TEST_P(urKernelGetInfoTest, NumArgs) {
+    auto property_name = UR_KERNEL_INFO_NUM_ARGS;
     size_t property_size = 0;
-    std::vector<char> property_value;
+
     ASSERT_SUCCESS(
```

Comment:
this will be marked optional in https://github.com/oneapi-src/unified-runtime/pull/1332/files#diff-ce9df9c86375bf045474cd36b47228c4ef58e2c49d21239c93fb5d7946fcc4c8R127 so you could should change this `ASSERT_SUCCESS` to `UUR_ASSERT_SUCCESS_OR_UNSUPPORTED` (or will actually be `ASSERT_SUCCESS_OR_OPTIONAL_QUERY` once that PR gets merged

Reply:
resolved now that PR has been merged